lib/internal/fs/promises.js

```diff
@@ -495,8 +494,9 @@ async function writeFile(path, data, options) {
 
 async function appendFile(path, data, options) {
   options = getOptions(options, { encoding: 'utf8', mode: 0o666, flag: 'a' });
-  options = copyObject(options);
-  options.flag = options.flag || 'a';
+  if (options.flag !== 'a') {
```

Comment:
It seems that this changes the previous behavior.
```suggestion
  if (!options.flag) {
```